src/interface/update.cpp

```diff
@@ -64,45 +68,98 @@ TaskStatus FluxDivergence(std::shared_ptr<Container<Real>> &in,
   return TaskStatus::complete;
 }
 
-void UpdateContainer(std::shared_ptr<Container<Real>> &in,
-                     std::shared_ptr<Container<Real>> &dudt_cont, const Real dt,
-                     std::shared_ptr<Container<Real>> &out) {
+TaskStatus FluxDivergenceMesh(std::shared_ptr<MeshData<Real>> &in_obj,
+                              std::shared_ptr<MeshData<Real>> &dudt_obj) {
+  const IndexDomain interior = IndexDomain::interior;
+
+  std::vector<MetadataFlag> flags({Metadata::Independent});
+  auto vin = in_obj->PackVariablesAndFluxes(flags);
+  auto dudt = dudt_obj->PackVariables(flags);
+  const IndexRange ib = vin.cellbounds.GetBoundsI(interior);
+  const IndexRange jb = vin.cellbounds.GetBoundsJ(interior);
+  const IndexRange kb = vin.cellbounds.GetBoundsK(interior);
+
+  const int ndim = vin.GetNdim();
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "FluxDivergenceMesh", DevExecSpace(), 0, vin.GetDim(5) - 1, 0,
```

Comment:
Why do we need to specify `DEFAULT_LOOP_PATTERN` here? Shouldn't it automatically know?

Reply:
Because this is a `parthenon::par_for` and not a `pmb->par_for`.
Currently the abstraction only provides the "flexible" version and we set all the default in the MeshBlock member functions.